extensions/libarchive/ManipulateArchive.cpp

```diff
@@ -0,0 +1,309 @@
+            if ((*it).entryName == targetEntry) {
+                logger_->log_info("ManipulateArchive found entry %s to move", targetEntry.c_str());
+                moveEntry = *it;
+                it = archiveMetadata.entryMetadata.erase(it);
+                found = true;
+                break;
+            } else {
+                it++;
+            }
+        }
+
+        if (found) {
+            moveEntry.entryName = destination;
+
+            // Update metadata
+            if (after != "") {
+                for (auto it = archiveMetadata.entryMetadata.begin();;) {
+                    if (it == archiveMetadata.entryMetadata.end()) {
+                        logger_->log_info("ManipulateArchive could not find entry %s to move entry after, so entry will be appended to end of archive", after.c_str());
+                        archiveMetadata.entryMetadata.insert(it, moveEntry);
+                        break;
+                    }
+
+                    if ((*it).entryName == after) {
+                        logger_->log_info("ManipulateArchive found entry %s to move entry after", after.c_str());
+                        it++;
+                        archiveMetadata.entryMetadata.insert(it, moveEntry);
+                        break;
+                    } else {
+                        it++;
+                    }
+                }
+            } else {
+                for (auto it = archiveMetadata.entryMetadata.begin();;) {
+                    if (it == archiveMetadata.entryMetadata.end()) {
+                        logger_->log_info("ManipulateArchive could not find entry %s to move entry before, so entry will be appended to end of archive", before.c_str());
+                        archiveMetadata.entryMetadata.insert(it, moveEntry);
+                        break;
+                    }
+
+                    if ((*it).entryName == before) {
+                        logger_->log_info("ManipulateArchive found entry %s to move entry before", before.c_str());
+                        archiveMetadata.entryMetadata.insert(it, moveEntry);
+
+                        break;
+                    } else {
+                        it++;
+                    }
+                }
+            }
+        } else {
+            logger_->log_info("ManipulateArchive could not find entry %s to move", targetEntry.c_str());
+        }
+    }
+
+    // Perform operation: TOUCH
+    if (operation == OPERATION_TOUCH) {
+        FocusArchiveEntry::ArchiveEntryMetadata touchEntry;
+        touchEntry.entryName = targetEntry;
+        touchEntry.entryType = AE_IFREG;
+        touchEntry.entrySize = 0;
+        touchEntry.entryMTime = time(nullptr);
+        touchEntry.entryMTimeNsec = 0;
+        touchEntry.entryUID = 0;
+        touchEntry.entryGID = 0;
+        touchEntry.entryPerm = 0777;
+
+        // Update metadata
+        if (after != "") {
```

Comment:
All of these loops are essentially the same thing, can we combine them into a function and reuse the code?

Reply:
Indeed. That's the next step; the first was getting the other processor merged, and then get this building on the ever-so-picky Xcode.